### ISSUE TITLE
Freeze when trying to specify port number in FTP cloud storage

### ISSUE THREAD

**Issue by pbienst** [koreader/koreader#7688]
When trying to specify a port number for the ftp cloud storage (e.g. ftp://myserver.com:6666), KOReader hangs when trying to connect, and I have to hard reset the device...

**Comment by NiLuJe** [comment]
Please respect the issue template.

**Comment by NiLuJe** [comment]
In particular: *which* device, ùwhich* KOReader version, *which* cloud provider.

And verbose debug logs might be mildly helpful, too.

As well as an idea of how long you waited before resorting to a hard reboot...

**Comment by Frenzie** [comment]
Incidentally I don't see a timeout here at a quick glance, so it might potentially be allowed to block indefinitely?
https://github.com/koreader/koreader/blob/e6027313e97138c21f014f708b6201bc4a64c350/frontend/apps/cloudstorage/ftpapi.lua

**Comment by NiLuJe** [comment]
I'm unfamiliar with LuaSocket's FTP module, but I assume it probably has a similar default "socket-level" timeout of 60s (an no upper timeout, but also no way to trip one anyway).

**Comment by NiLuJe** [comment]
Yup: https://github.com/diegonehab/luasocket/blob/5b18e475f38fcf28429b1cc4b17baee3b9793a62/src/ftp.lua#L24

**Comment by pbienst** [comment]
I waited quite some time before rebooting (probably 30 sec) or so, but never 60s, so I tried that again. This time it does recover with an error that it can't fetch the folder contents. This is strange, because different FTP servers with the standard port work fine using koreader, and using a PC to FTP into the server with the non-standard port also works fine.

This leads me to believe that koreader does not properly take the port number into account. Is ftp://myserver.com:6666 the correct syntax here?

BTW, crash logs don't show anything FTP related:

 It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 05/15/21-20:10:52
 [*] Version: v2021.04

ffi.load: blitbuffer
ffi.load: SDL2
ffi.load (warning): libSDL2.so: cannot open shared object file: No such file or directory
ffi.load: libSDL2-2.0.so
ffi.load (warning): libSDL2-2.0.so: cannot open shared object file: No such file or directory
ffi.load: libSDL2-2.0.so.0
ffi.load (warning): libSDL2-2.0.so.0: cannot open shared object file: No such file or directory
SDL2 not loaded:	./setupkoenv.lua:27: Not able to load dynamic library: libSDL2-2.0.so.0
05/15/21-20:10:52 INFO  initializing for device Kobo_nova
05/15/21-20:10:52 INFO  framebuffer resolution: {
    ["w"] = 1072,
    ["h"] = 1448,
}
[ko-input] Forked off fake event generator (pid: 1887)
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
PING 192.168.2.1 (192.168.2.1): 56 data bytes
64 bytes from 192.168.2.1: seq=0 ttl=64 time=5.800 ms

--- 192.168.2.1 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 5.800/5.800/5.800 ms
PING 192.168.2.1 (192.168.2.1): 56 data bytes
64 bytes from 192.168.2.1: seq=0 ttl=64 time=2.970 ms

--- 192.168.2.1 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 2.970/2.970/2.970 ms
05/15/21-20:10:54 INFO  Loading plugins from directory: plugins
05/15/21-20:10:54 INFO  Plugin  plugins/autofrontlight.koplugin/main.lua  has been disabled.
ffi.load: sqlite3
05/15/21-20:10:54 INFO  Plugin  plugins/hello.koplugin/main.lua  has been disabled.
05/15/21-20:10:55 INFO  Plugin  plugins/autostandby.koplugin/main.lua  has been disabled.
05/15/21-20:10:55 INFO  Plugin  plugins/coverimage.koplugin/main.lua  has been disabled.
05/15/21-20:10:55 INFO  FM loaded plugin SSH at plugins/SSH.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin autosuspend at plugins/autosuspend.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin backgroundrunner at plugins/backgroundrunner.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin batterystat at plugins/batterystat.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin calibre at plugins/calibre.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin coverbrowser at plugins/coverbrowser.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin docsettingtweak at plugins/docsettingtweak.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin evernote at plugins/evernote.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin gestures at plugins/gestures.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin goodreads at plugins/goodreads.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin keepalive at plugins/keepalive.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin movetoarchive at plugins/movetoarchive.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin newsdownloader at plugins/newsdownloader.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin opds at plugins/opds.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin perceptionexpander at plugins/perceptionexpander.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin profiles at plugins/profiles.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin qrclipboard at plugins/qrclipboard.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin readtimer at plugins/readtimer.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin send2ebook at plugins/send2ebook.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin statistics at plugins/statistics.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin systemstat at plugins/systemstat.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin terminal at plugins/terminal.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin texteditor at plugins/texteditor.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin timesync at plugins/timesync.koplugin
05/15/21-20:10:55 INFO  FM loaded plugin wallabag at plugins/wallabag.koplugin
PING 192.168.2.1 (192.168.2.1): 56 data bytes
64 bytes from 192.168.2.1: seq=0 ttl=64 time=5.619 ms

--- 192.168.2.1 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 5.619/5.619/5.619 ms
05/15/21-20:10:55 INFO  Wi-Fi successfully restored (after 1 iterations)!
05/15/21-20:11:03 INFO  GestureDetector:probeClockSource: Touch event timestamps appear to use CLOCK_REALTIME
PING 192.168.2.1 (192.168.2.1): 56 data bytes
64 bytes from 192.168.2.1: seq=0 ttl=64 time=6.859 ms

--- 192.168.2.1 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 6.859/6.859/6.859 ms
05/15/21-20:13:27 INFO  quitting uimanager
